spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDefinitionController.java

```diff
@@ -116,13 +117,22 @@ public void destroyTask(@PathVariable("name") String name) {
 	 *
 	 * @param pageable  page-able collection of {@code TaskDefinitionResource}.
 	 * @param assembler assembler for the {@link TaskDefinition}
+	 * @param search optional search parameter
 	 * @return a list of task definitions
 	 */
 	@RequestMapping(value="", method = RequestMethod.GET)
 	@ResponseStatus(HttpStatus.OK)
-	public PagedResources<TaskDefinitionResource> list(Pageable pageable,
+	public PagedResources<TaskDefinitionResource> list(Pageable pageable, @RequestParam(required=false) String search,
 			PagedResourcesAssembler<TaskDefinition> assembler) {
-		return assembler.toResource(repository.findAll(pageable), taskAssembler);
+
+		if (search != null) {
+			final SearchPageable searchPageable = new SearchPageable(pageable, search);
+			searchPageable.addColumns("DEFINITION_NAME", "DEFINITION");
```

Comment:
We don't otherwise know what to search over. I made it configurable on the controller-level so we could potentially add more finer-grained searches. Maybe we don't want to search across the definitions (only the names). 

Although, I am not particularly fond of providing exact database columns names (properties would be better IMO).